openquake/smt/comparison/utils_gmpes.py

```diff
@@ -0,0 +1,331 @@
+                                   direction='positive', hdist=maxR, step=step,
+                                   site_props=props)
+    
+    param = dict(imtls={})
+    cm = ContextMaker(trt, [gmpe], param)
+    
+    ctxs = list(cm.get_ctx_iter([rup], sites)) 
+    ctxs = ctxs[0]
+    ctxs.occurrence_rate = 0.0
+    
+    mean, std = gmpe.get_mean_and_stddevs(ctxs, ctxs, ctxs, imt, [StdDev.TOTAL])
+    distances = ctxs.rrup
+    
+    return mean, std, distances
+
+def AttCurvesESHM20(gmpe,depth,mag,aratio,strike,dip,rake,Vs30,Z1,Z25,maxR,
+                    step,imt,ztor,region):    
+    
+    trt = gmpe.DEFINED_FOR_TECTONIC_REGION_TYPE
+    rup = get_rupture(0.0, 0.0, depth, WC1994(), mag=mag, aratio=aratio,
+                      strike=strike, dip=dip, rake=rake, trt=trt, ztor=ztor)
+    props = {'vs30': Vs30, 'z1pt0': Z1, 'z2pt5': Z25, 'backarc': False,
+             'vs30measured': True,'region': numpy.array(region)}                
+    sites = get_sites_from_rupture(rup, from_point='TC', toward_azimuth=90,
+                                   direction='positive', hdist=maxR,
+                                   step=step, site_props=props)
+    param = dict(imtls={})
+    cm = ContextMaker(trt, [gmpe], param)
+    
+    ctxs = list(cm.get_ctx_iter([rup], sites))          
+    ctxs = ctxs[0]
+    ctxs.occurrence_rate = 0.0
+            
+    mean, std = gmpe.get_mean_and_stddevs(ctxs, ctxs, ctxs, imt, [StdDev.TOTAL])
+    distances = ctxs.rrup
+    
+    return mean, std, distances
+
+def _get_Z1(Vs30,region):
+    # choose among: region= 0 for global; 1 for California; 2 for Japan; 3 for China; 4 for Italy; 5 for Turkey (locally = 3); 6 for Taiwan (locally = 0)      
+    if region == 2: # in California and non-Japan region
+        Z1 = numpy.exp(-5.23/2*numpy.log((Vs30**2+412.39**2)/(1360**2+412.39**2)))
+    else:
+        Z1 = numpy.exp(-7.15/4*numpy.log((Vs30**4+570.94**4)/(1360**4+570.94**4)))
+    return Z1
+
+def _get_Z25(Vs30,region):
+    # choose among: region= 0 for global; 1 for California; 2 for Japan; 3 for China; 4 for Italy; 5 for Turkey (locally = 3); 6 for Taiwan (locally = 0)      
+    if region == 2: # in Japan
+        Z25 = numpy.exp(5.359 - 1.102 * numpy.log(Vs30))
+        Z25A_default = numpy.exp(7.089 - 1.144 * numpy.log(1100))
+    else:
+        Z25 = numpy.exp(7.089 - 1.144 * numpy.log(Vs30))
+        Z25A_default = numpy.exp(7.089 - 1.144 * numpy.log(1100))           
+    return Z25
+
+def _param_gmpes(gmpes, strike, dip, depth, aratio, rake):
+    # specific assumptions when the param are not available from the sources
+    if strike == -999: 
+        strike_s = 0
+    else:
+        strike_s = strike
+
+    if dip == -999:
+        if rake == 0:
+            dip_s = 90 # strike slip
+        else:
+            dip_s = 45 # reverse or normal fault 
+    else:
+        dip_s = dip
+                      
+    if depth == -999:
+        if 'Inter' in gmpes:
```

Comment:
I guess you are assuming that a GMM with Inter in the name is for Interface. I think it would be better to define it explicitly.

Reply:
the "Inter" bit is read directly from the name of the GMPE.. so we avoid to define an additional parameter: for what I have seen all GMMs for interface have "Inter" within the name in OQ...

Reply:
These conditional statements now get the tectonic region type for each gmpe as a string and check against inter and intra tectonic region codes. I checked using some subduction gmpes including Abrahahmson and Gulerce 2020 subduction gmpe and it works this way I believe